sql-jdbc/src/main/java/org/opensearch/jdbc/ResultSetImpl.java

```diff
@@ -98,6 +99,26 @@ public ResultSetImpl(StatementImpl statement, List<? extends ColumnDescriptor> c
 
             List<Row> rows = getRowsFromDataRows(dataRows);
 
+            for (int i = 0; i < columnDescriptors.size(); i ++) {
```

Comment:
you can use array streams to filter the columns by type (DATE|TIME|DATETIME), and then get the max value row for that column. 
Then you don't have to include a double for loop

Reply:
Should we check to see if the precious is already defined in the metadata beforehand?

Reply:
> Should we check to see if the Precision is already defined in the metadata beforehand?

No, I change the constructor.

Reply:
> Then you don't have to include a double for loop

I have to loop through all rows/columns anyway

Reply:
Can we talk about this approach? I see what you're doing, but I have some thoughts about the pros/cons.

Reply:
For sure!